wagtail/wagtailcore/fields.py

```diff
@@ -35,8 +45,13 @@ def value_from_datadict(self, data, files, name):
 
 
 class RichTextField(models.TextField):
+
+    def __init__(self, hallo_config=None, **kwargs):
```

Comment:
I'm not sure that field should know how to display data: django form fields is about processing data (validation, conversion, etc). I think only widget should know about hallo plugins configuration.

Maybe we should just provide possibility to configure Hallo in `RichTextArea` widget and provide possibility to change widget?

It may look somehow like this:
```python
class BookPage(Page):
    book_text = RichTextField(
        widget=RichTextArea(hallo_plugins={
            'halloheadings': {...},
            ...
        })
    )
```